### DISCUSSION TITLE
Unable to add Recaptcha field within form or fields sections

### DISCUSSION THREAD

**Discussion by zizther** [solspace/craft-freeform#387] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**
I am attempting to add a new Recaptcha field which is currently not available from the fields list.
When I click the `Save` button in the Form field sidebar or the Field section the page does not do anything, no console.log are shown.
I have checked the log files and there are no errors.

I am able to make any other field type besides the Recaptcha field.

I have Google Recpatcha V3 settings added to the setting page.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.46
- Freeform Version: 3.13.14
- Freeform Edition: Pro
- Fresh Install or Upgrade: Fresh install

**Comment by kjmartens** [comment, marked as answer]
Hi @zizther,

I believe this is a misunderstanding of how reCAPTCHA v3 works. It does not appear as a field in any way. **v2 Checkbox** will appear as a field, and it shows up in the top left "Special Fields" area of the form builder to be dragged into the form layout (not created like a normal field).

Here's more information for context... [reCAPTCHA is available in 3 different options](https://docs.solspace.com/craft/freeform/v3/overview/spam-protection.html#captchas):

- **reCAPTCHA v2 Checkbox** is likely the reCAPTCHA you may be most familiar with seeing on the internet. In this case, once configured for your site, you drag and drop a special Captcha field into the form layout for the forms you wish to have it enabled for. All users submitting your form must check off the reCAPTCHA checkbox, and in some cases, users will be presented the puzzle to solve to submit the form successfully.
- **reCAPTCHA v2 Invisible** is similar to reCAPTCHA v2 Checkbox, but it will not be visible to the user filling out the form, aside from a small reCAPTCHA logo in the bottom right corner of the page. Most users will not even know it's automatically validating them, but like the reCAPTCHA v2 Checkbox described above, some users will be presented a modal on the page with the puzzle to solve when they click the submit button.
   - This automatically applies to every form unless you uncheck the "Enable Captchas" checkbox in the form settings in the form builder or disabled at template level.
- **reCAPTCHA v3** is never visible to the user filling out the form, aside from a small reCAPTCHA logo in the bottom right corner of the page, and users are never presented any puzzle to solve, etc. Instead, users are automatically validated by Google based on a score assigned to them. You have control over that score threshold inside Freeform settings. We suggest starting with something like `0.5` and see how that goes (where a `0.0` score means that it's almost certain a bot and a `1.0` score means it's a user). If the user does not pass this threshold, the submission will follow the behavior you set out for it in the settings (send to spam folder / reject it entirely / show an error to the user, etc). The user has no ability to validate themselves like in reCAPTCHA v2.
   - This automatically applies to every form unless you uncheck the "Enable Captchas" checkbox in the form settings in the form builder or disabled at template level.

**Comment by zizther** [comment]
@kjmartens Thanks for getting back on this one.
Would it not make more sense to remove recaptcha from the fields list if you have V3 selected or show a message if trying to add it, just so that it is clear?

**Reply by kjmartens** [reply]
Sorry about that. It definitely shouldn't be showing up in there. Perhaps a recent release of Freeform had this begin to show up there. We'll get it fixed.
